Image Optimization using the new package - spatie/image-optimizer (#11763)
pimcore/pimcore: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

  instead of `pimcore_core`. If you want to handle image optimize messages, then it is required to add specific option `pimcore_image_optimize` to the command `bin/console messenger:consume pimcore_core pimcore_maintenance pimcore_image_optimize`. Also run command `bin/console messenger:consume pimcore_core` before the upgrade, so that ImageOptimize messages on the queue gets consumed.
- **Important**: [Object bricks] A call to the object brick´s getter method no longer returns object bricks marked for deletion. 
  To restore the original behavior pass "true" to the getter method´s `$includeDeletedBricks` argument. 

> - [Spatie/image-optimizer instead of custom] Custom image optimizer (for thumbnail images) is deprecated and a new image optimizer Spatie/image optimizer is introduced.

Reviewer: This sentence is somehow broken...

  instead of `pimcore_core`. If you want to handle image optimize messages, then it is required to add specific option `pimcore_image_optimize` to the command `bin/console messenger:consume pimcore_core pimcore_maintenance pimcore_image_optimize`. Also run command `bin/console messenger:consume pimcore_core` before the upgrade, so that ImageOptimize messages on the queue gets consumed.
- **Important**: [Object bricks] A call to the object brick´s getter method no longer returns object bricks marked for deletion. 
  To restore the original behavior pass "true" to the getter method´s `$includeDeletedBricks` argument. 

> - [Image Optimizer] Image Optimizer services (e.g. PngCrushOptimizer, JpegoptimOptimizer etc.) are deprecated and will be

Reviewer 2: We should also mention that the existing optimizers were disabled by default and we should also describe here how to re-enable them again if necessary.

  instead of `pimcore_core`. If you want to handle image optimize messages, then it is required to add specific option `pimcore_image_optimize` to the command `bin/console messenger:consume pimcore_core pimcore_maintenance pimcore_image_optimize`. Also run command `bin/console messenger:consume pimcore_core` before the upgrade, so that ImageOptimize messages on the queue gets consumed.
- **Important**: [Object bricks] A call to the object brick´s getter method no longer returns object bricks marked for deletion. 
  To restore the original behavior pass "true" to the getter method´s `$includeDeletedBricks` argument. 
- [Image Optimizer] Image Optimizer services (e.g. PngCrushOptimizer, JpegoptimOptimizer etc.) are deprecated and will be
  removed in Pimcore 11. Use Pimcore\Image\Optimizer\SpatieImageOptimizer service instead. 

>   Currently, the existing optimizers are disabled and if you still want to use them, then please re-enable it as follows. In the image_optimizers.yaml file, add the tag " { name: pimcore.image.optimizer }" under the required service definitions. For eg: if you want to re-enable the CjpegOptimizer, you can add the tag { name: pimcore.image.optimizer } under the service definition of CjpegOptimizer ( Pimcore\Image\Optimizer\JpegoptimOptimizer ).

Reviewer: ```suggestion
  Currently, the existing optimizers are disabled. If you still want to use them, then please re-enable them as follows: in the `image_optimizers.yaml` file, add the tag `{ name: pimcore.image.optimizer }` under the required service definitions. For example: if you want to re-enable the `CjpegOptimizer`, you can add the tag `{ name: pimcore.image.optimizer }` under the service definition of `CjpegOptimizer` (which is: `Pimcore\Image\Optimizer\JpegoptimOptimizer`).
```

>   Currently, the existing optimizers are disabled. If you still want to use them, then please re-enable them as follows: in the `image_optimizers.yaml` file, add the tag `{ name: pimcore.image.optimizer }` under the required service definitions. For example: if you want to re-enable the `CjpegOptimizer`, you can add the tag `{ name: pimcore.image.optimizer }` under the service definition of `CjpegOptimizer` (which is: `Pimcore\Image\Optimizer\JpegoptimOptimizer`).

Reviewer 2: ```suggestion
  Currently, the existing optimizers are disabled. If you still want to use them, then please re-enable them by tagging the services accordingly (in your `services.yaml`):
  
```yaml
    Pimcore\Image\Optimizer\CjpegOptimizer:
        tags:
            - { name: pimcore.image.optimizer }

    Pimcore\Image\Optimizer\JpegoptimOptimizer:
        tags:
            - { name: pimcore.image.optimizer }

    Pimcore\Image\Optimizer\PngCrushOptimizer:
        tags:
            - { name: pimcore.image.optimizer }
            
```
```
